lib/Sema/DerivedConformanceCodable.cpp

```diff
@@ -300,9 +300,17 @@ static CodingKeysValidity hasValidCodingKeysEnum(TypeChecker &tc,
   if (!tc.conformsToProtocol(codingKeysType, codingKeyProto,
                              target->getDeclContext(),
                              ConformanceCheckFlags::Used)) {
-    tc.diagnose(codingKeysTypeDecl->getLoc(),
-                diag::codable_codingkeys_type_does_not_conform_here,
+    // If CodingKeys is a typealias which doesn't point to a valid nominal type,
```

Comment:
The codable_codingkeys_type_is_not_an_enum_here diagnostic seems misleading, because we never actually check that codingKeysDecl is an enum.

Reply:
We do that below on line 318